### ISSUE TITLE
[Bug Report] Android MQTT Connection is Lost when sent a message from Cloud

### ISSUE THREAD

**Issue by Abdul-Moiz-Hassan** [Azure/azure-iot-sdk-java#1727]
# Context

- **OS and version used:** <VERSION> Android 11 (API 30)
- **Java runtime used:** <VERSION>API Level 30
- **SDK version used:** <VERSION> Latest `com.microsoft.azure.sdk.iot:iot-device-client:2.2.0`

## Description of the issue
I have been using Azure IoT Hub primarily for Device-to-Cloud (D2C) messaging. Recently, I decided to switch to Cloud-to-Device (C2D) messaging. To facilitate this, I upgraded my Azure IoT Hub tier from Basic to Standard.

However, when I attempt to send any message from the cloud using the Azure Web Dashboard, the client gets disconnected and these disconnections are accompanied by exceptions. Initially, I suspected that I might have misconfigured some settings, so I tested using the provided Android Sample app located at iothub\device\iot-device-samples\android-sample.

To my surprise, the issue persisted. The client connects successfully and D2C messages are sent without a hitch. However, as soon as a message is sent to the device (C2D), the client disconnects and fails to reconnect, even after restarting the application.

I have gone through the documentation and reviewed previous issues, but none of them seem to address this problem. I'm at a loss as to what might be causing this issue. Any guidance or troubleshooting tips would be greatly appreciated.

Here is how I am sending C2D messages from Azure Web Dashboard:

![image](https://github.com/Azure/azure-iot-sdk-java/assets/40466350/77a276c2-a85a-437a-ae73-2d95ff11341c)

## Code sample exhibiting the issue

As I mentioned this issue occurs in the sample anroid app, withouth need to change anything (I did changed the `connString` ofcourse).
```
        client = new DeviceClient(connString, protocol);

        try
        {
            client.open(true);

            if (protocol == IotHubClientProtocol.MQTT)
            {
                MessageCallbackMqtt callback = new MessageCallbackMqtt();
                Counter counter = new Counter(0);
                client.setMessageCallback(callback, counter);
            }
            else
            {
                MessageCallback callback = new MessageCallback();
                Counter counter = new Counter(0);
                client.setMessageCallback(callback, counter);
            }
```

## Console log of the issue
```
---------------------------- PROCESS STARTED (16440) for package com.iothub.azure.microsoft.com.androidsample ----------------------------
2023-07-11 12:57:38.762 16440-16440 Perf                    com...e.microsoft.com.androidsample  I  Connecting to perf service.
2023-07-11 12:57:38.777 16440-16440 NetworkSecurityConfig   com...e.microsoft.com.androidsample  D  No Network Security Config specified, using platform default
2023-07-11 12:57:38.778 16440-16440 NetworkSecurityConfig   com...e.microsoft.com.androidsample  D  No Network Security Config specified, using platform default
2023-07-11 12:57:38.874 16440-16440 Applicatio...ageManager com...e.microsoft.com.androidsample  D  hasSystemFeatrue android.software.picture_in_picture com.iothub.azure.microsoft.com.androidsample
2023-07-11 12:57:38.983 16440-16440 m.androidsampl          com...e.microsoft.com.androidsample  W  Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (greylist, reflection, allowed)
2023-07-11 12:57:38.985 16440-16440 m.androidsampl          com...e.microsoft.com.androidsample  W  Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (greylist, reflection, allowed)
2023-07-11 12:57:39.039 16440-16440 *ntialBackoffWithJitter com...e.microsoft.com.androidsample  I  NOTE: A new instance of ExponentialBackoffWithJitter has been created with the following properties. Retry Count: 2147483647, Min Backoff Interval: 100, Max Backoff Interval: 10000, Max Time Between Retries: 100, Fast Retry Enabled: true
2023-07-11 12:57:39.054 16440-16440 *ntialBackoffWithJitter com...e.microsoft.com.androidsample  I  NOTE: A new instance of ExponentialBackoffWithJitter has been created with the following properties. Retry Count: 2147483647, Min Backoff Interval: 100, Max Backoff Interval: 10000, Max Time Between Retries: 100, Fast Retry Enabled: true
2023-07-11 12:57:40.897 16440-16440 DeviceClient            com...e.microsoft.com.androidsample  I  Device client opened successfully
2023-07-11 12:57:42.231 16440-16471 AdrenoGLES-0            com...e.microsoft.com.androidsample  I  QUALCOMM build                   : 2bd16eb911, I3687d7cf91
                                                                                                    Build Date                       : 02/10/21
                                                                                                    OpenGL ES Shader Compiler Version: EV031.32.02.08
                                                                                                    Local Branch                     : 
                                                                                                    Remote Branch                    : 
                                                                                                    Remote Branch                    : 
                                                                                                    Reconstruct Branch               : 
2023-07-11 12:57:42.231 16440-16471 AdrenoGLES-0            com...e.microsoft.com.androidsample  I  Build Config                     : S P 10.0.7 AArch64
2023-07-11 12:57:42.231 16440-16471 AdrenoGLES-0            com...e.microsoft.com.androidsample  I  Driver Path                      : /vendor/lib64/egl/libGLESv2_adreno.so
2023-07-11 12:57:42.238 16440-16471 AdrenoGLES-0            com...e.microsoft.com.androidsample  I  PFP: 0x016ee195, ME: 0x00000000
2023-07-11 12:57:51.754 16440-16440 System.out              com...e.microsoft.com.androidsample  I  {"temperature":24.79948858594452,"humidity":48.22031741088057}
2023-07-11 12:57:52.092 16440-16501 System.out              com...e.microsoft.com.androidsample  I  IoT Hub responded to message 04419c19-4639-4652-84ec-1217f59d3273 with status OK
2023-07-11 12:57:53.972 16440-16440 System.out              com...e.microsoft.com.androidsample  I  {"temperature":28.90296054278825,"humidity":34.57995521291}
2023-07-11 12:57:54.367 16440-16501 System.out              com...e.microsoft.com.androidsample  I  IoT Hub responded to message 151afac3-c0e2-4bb4-8297-ec19343f7c2e with status OK
2023-07-11 12:58:02.802 16440-16440 System.out              com...e.microsoft.com.androidsample  I  {"temperature":22.696700063641284,"humidity":37.86135492269476}
2023-07-11 12:58:03.253 16440-16501 System.out              com...e.microsoft.com.androidsample  I  IoT Hub responded to message 25198741-7ff5-448a-9b15-dab48a32cb05 with status OK
2023-07-11 12:58:48.090 16440-16495 Mqtt                    com...e.microsoft.com.androidsample  W  Mqtt connection lost
                                                                                                    MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
2023-07-11 12:58:48.098 16440-16495 IotHubTransport         com...e.microsoft.com.androidsample  W  Updating transport status to new status DISCONNECTED_RETRYING with reason COMMUNICATION_ERROR
                                                                                                    com.microsoft.azure.sdk.iot.device.transport.ProtocolException: Mqtt connection lost
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.exceptions.PahoExceptionTranslator.convertToMqttException(PahoExceptionTranslator.java:42)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.connectionLost(Mqtt.java:339)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.connectionLost(CommsCallback.java:304)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.ClientComms.shutdownConnection(ClientComms.java:441)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228) 
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
2023-07-11 12:58:51.984 16440-16503 Mqtt                    com...e.microsoft.com.androidsample  W  Encountered exception while sending MQTT SUBSCRIBE packet for topic devices/andriod-test/messages/devicebound/#
                                                                                                    MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
2023-07-11 12:58:51.992 16440-16802 Mqtt                    com...e.microsoft.com.androidsample  W  Mqtt connection lost
                                                                                                    MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
2023-07-11 12:58:51.993 16440-16503 IotHubTransport         com...e.microsoft.com.androidsample  W  Failed to close and re-open the iot hub transport connection, checking if another retry attempt should be made
                                                                                                    com.microsoft.azure.sdk.iot.device.transport.ProtocolException: Unable to subscribe to topic :devices/andriod-test/messages/devicebound/#
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.exceptions.PahoExceptionTranslator.convertToMqttException(PahoExceptionTranslator.java:42)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.subscribe(Mqtt.java:280)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttMessaging.start(MqttMessaging.java:66)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.open(MqttIotHubConnection.java:286)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.openConnection(IotHubTransport.java:1309)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.singleReconnectAttempt(IotHubTransport.java:1411)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.reconnect(IotHubTransport.java:598)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubReconnectTask.run(IotHubReconnectTask.java:69)
                                                                                                    	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
                                                                                                    	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:307)
                                                                                                    	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:302)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
2023-07-11 12:58:52.001 16440-16503 IotHubTransport         com...e.microsoft.com.androidsample  E  Reconnection was abandoned due to encountering a non-retryable exception
                                                                                                    com.microsoft.azure.sdk.iot.device.transport.ProtocolException: Unable to subscribe to topic :devices/andriod-test/messages/devicebound/#
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.exceptions.PahoExceptionTranslator.convertToMqttException(PahoExceptionTranslator.java:42)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.subscribe(Mqtt.java:280)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttMessaging.start(MqttMessaging.java:66)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.open(MqttIotHubConnection.java:286)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.openConnection(IotHubTransport.java:1309)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.singleReconnectAttempt(IotHubTransport.java:1411)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.reconnect(IotHubTransport.java:598)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubReconnectTask.run(IotHubReconnectTask.java:69)
                                                                                                    	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
                                                                                                    	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:307)
                                                                                                    	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:302)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
2023-07-11 12:58:52.005 16440-16503 IotHubTransport         com...e.microsoft.com.androidsample  W  Updating transport status to new status DISCONNECTED with reason COMMUNICATION_ERROR
                                                                                                    com.microsoft.azure.sdk.iot.device.transport.ProtocolException: Unable to subscribe to topic :devices/andriod-test/messages/devicebound/#
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.exceptions.PahoExceptionTranslator.convertToMqttException(PahoExceptionTranslator.java:42)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.subscribe(Mqtt.java:280)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttMessaging.start(MqttMessaging.java:66)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.open(MqttIotHubConnection.java:286)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.openConnection(IotHubTransport.java:1309)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.singleReconnectAttempt(IotHubTransport.java:1411)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubTransport.reconnect(IotHubTransport.java:598)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.IotHubReconnectTask.run(IotHubReconnectTask.java:69)
                                                                                                    	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
                                                                                                    	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:307)
                                                                                                    	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:302)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
                                                                                                    	at java.lang.Thread.run(Thread.java:923)
                                                                                                    Caused by: MqttException (0) - java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:228)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
                                                                                                    Caused by: java.lang.ArrayIndexOutOfBoundsException: length=1; index=1
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.assignPropertiesToMessage(Mqtt.java:476)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.constructMessage(Mqtt.java:442)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.receive(Mqtt.java:306)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.onMessageArrived(MqttIotHubConnection.java:458)
                                                                                                    	at com.microsoft.azure.sdk.iot.device.transport.mqtt.Mqtt.messageArrived(Mqtt.java:364)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.deliverMessage(CommsCallback.java:519)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.handleMessage(CommsCallback.java:417)
                                                                                                    	at org.eclipse.paho.client.mqttv3.internal.CommsCallback.run(CommsCallback.java:214)
                                                                                                    	at java.lang.Thread.run(Thread.java:923) 
```

**Comment by timtay-microsoft** [comment]
This appears to be a duplicate of #1722

**Comment by timtay-microsoft** [comment]
The TL;dr is that the service team accidentally changed some behavior and are working to un-break it. You can file a ticket through the Azure portal to have them revert this change to your hub, or you can ensure that every property field on your c2d messages has a value

**Comment by Abdul-Moiz-Hassan** [comment]
My apologies for any confusion. I mistakenly created a duplicate issue (#1722) due to an oversight on my part.

In my initial search, I was primarily focused on occurrences of the `ProtocolException` error, and it seems I inadvertently overlooked the `ArrayIndexOutOfBoundsException`. Here's the exception I was initially looking into:
```
com.microsoft.azure.sdk.iot.device.transport.ProtocolException: Unable to subscribe to topic :devices/andriod-test/messages/devicebound/#
at com.microsoft.azure.sdk.iot.device.transport.mqtt.exceptions.PahoExceptionTranslator.convertToMqttException(PahoExceptionTranslator.java:42)
```

Anyways I did try adding some properties in my C2D message, but I am still having that same `ArrayIndexOutOfBoundsException`. I added some system properties like this:

![image](https://github.com/Azure/azure-iot-sdk-java/assets/40466350/6f89269c-1306-4ef7-b597-30585a8e978b)


> In the meantime, you can file a ticket through the Azure portal to have them revert this change to your specific hub

I have also filed a ticket to revert changes for my IoT hub. Hopefully, the issue will be fixed from the service end.
